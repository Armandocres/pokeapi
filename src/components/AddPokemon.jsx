import React, { useState } from "react";
import PropTypes from "prop-types";

const AddPokemon = ({ setpoke }) => {
	const [inputValue, setinputValue] = useState("");

	const handleInput = (e) => {
		setinputValue(e.target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		if (inputValue.trim().length > 2) {
			setpoke(inputValue);
			setinputValue("");
		}
	};

	return (
		<form onSubmit={handleSubmit} className="form">
			<label htmlFor="pokemon">Busca a tu pokemon</label>
			<input type="text" value={inputValue} onChange={handleInput} />
		</form>
	);
};

AddPokemon.propTypes = {
	setpoke: PropTypes.func.isRequired,
};

export default AddPokemon;
