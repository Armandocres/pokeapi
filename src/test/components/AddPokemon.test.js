import React from 'react';
import { shallow } from 'enzyme';
import AddPokemon from '../../components/AddPokemon';

describe('Pruebas en componente AddPokemon', () => {
  const setPoke = jest.fn();
  let wrapper = shallow(<AddPokemon setpoke={setPoke} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddPokemon setpoke={setPoke} />);
  });

  test('Pruebas en Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe cambiarla caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';

    input.simulate('change', { target: { value: value }
    });
  });

  test('No debe de lanzarse el evento Submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect(setPoke).not.toHaveBeenCalled();
  });

  test('Debe llamar al setPoke y limpiar la caja de texto', () => {
    const value = 'Hola mundo';
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect(setPoke).toHaveBeenCalled();
    expect(wrapper.find('input').prop('value')).toBe('');
  });

});