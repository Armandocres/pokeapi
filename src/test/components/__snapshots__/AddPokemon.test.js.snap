// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente AddPokemon Pruebas en Snapshot 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <label
    htmlFor="pokemon"
  >
    Busca a tu pokemon
  </label>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
